### PR TITLE
refactor: Move document state help to /doc

### PR DESCRIPTION
**Pull request by jimfenton**

### CODE REVIEW

File: ietf/help/views.py [hunk 1]
```diff
@@ -1,30 +1,20 @@
...
-                
-    return render(request, 'help/state_index.html', {"types": types})
+    return redirect('/doc/help/state/', permanent = True)
 
 def state(request, doc, type=None):
     if type:
         streams = [ s.slug for s in StreamName.objects.all() ]
         if type in streams:
             type = "stream-%s" % type
     slug = "%s-%s" % (doc,type) if type else doc
```
Comment (rjsparks) [inline ietf/help/views.py:18]: else doc.type_id maybe?

File: ietf/doc/views_help.py [hunk 2]
```diff
@@ -9,6 +9,18 @@
...
+    types = StateType.objects.all()
+    names = [ type.slug for type in types ]
+    for type in types:
+        if "-" in type.slug and type.slug.split('-',1)[0] in names:
+            type.stategroups = None
+        else:
+            groups = StateType.objects.filter(slug__startswith=type.slug)
+            type.stategroups =  [ g.slug[len(type.slug)+1:] for g in groups if not g == type ] or ""
+                
+    return render(request, 'doc/state_index.html', {"types": types})
```
Comment (rjsparks) [inline ietf/doc/views_help.py:22]: the concept of `used` on State probably needs to be rendered here somewhere - maybe that can be done entirely in the templates.

File: ietf/help/views.py [hunk 3]
```diff
@@ -1,30 +1,20 @@
...
-    names = [ type.slug for type in types ]
-    for type in types:
-        if "-" in type.slug and type.slug.split('-',1)[0] in names:
-            type.stategroups = None
-        else:
-            groups = StateType.objects.filter(slug__startswith=type.slug)
-            type.stategroups =  [ g.slug[len(type.slug)+1:] for g in groups if not g == type ] or ""
-                
-    return render(request, 'help/state_index.html', {"types": types})
+    return redirect('/doc/help/state/', permanent = True)
```
Comment (rjsparks) [inline ietf/help/views.py:11]: This redirect could be moved directly into the URLs file, and the view code and test for it can be removed.

**Review by rjsparks — CHANGES_REQUESTED** [review]
Jim - this is great and gets most of the way there, but we need to touch a few things before bringing this into main.

File: ietf/help/views.py (hunk 1 above)
Comment (jimfenton) [reply ietf/help/views.py:18]: No, in this context `doc` is a simple string variable that comes from a portion of the URL. Maybe that's a poor choice of variable name (preexisting).

**Comment by jimfenton** [comment]
Also noticed that doc/help/state/draft-iana/ now returns a 404. /help/state/draft/iana/ didn't produce a very useful page before, but at least it didn't generate a 404.

**Comment by rjsparks** [comment]
> Also noticed that doc/help/state/draft-iana/ now returns a 404. /help/state/draft/iana/ didn't produce a very useful page before, but at least it didn't generate a 404.

Interesting catch - `draft-iana` isn't a document state type. I think that it passing through the URLs to date was a bug.

**Comment by rjsparks** [comment]
I've deleted the unused StateType in production using the admin.

File: ietf/doc/views_help.py (hunk 2 above)
Comment (rjsparks) [reply ietf/doc/views_help.py:22]: nevermind - the bit is badly used in the State table.

**Review by rjsparks — APPROVED** [review]

**Pull request merged**